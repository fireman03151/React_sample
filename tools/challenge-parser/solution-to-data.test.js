/* global describe it expect beforeEach */
const mockAST = require('./fixtures/challenge-html-ast.json');
const solutionToData = require('./solution-to-data');
const { isObject } = require('lodash');

describe('challengeSeed-to-data plugin', () => {
  const plugin = solutionToData();
  let file = { data: {} };

  beforeEach(() => {
    file = { data: {} };
  });

  it('returns a function', () => {
    expect(typeof plugin).toEqual('function');
  });

  it('adds a `solutions` property to `file.data`', () => {
    plugin(mockAST, file);
    expect('solutions' in file.data).toBe(true);
  });

  it('ensures that the `solutions` property is an array', () => {
    plugin(mockAST, file);
    expect(Array.isArray(file.data.solutions)).toBe(true);
  });

  it('each entry in the `solutions` array is an object', () => {
    plugin(mockAST, file);

    expect(file.data.solutions.every(el => isObject(el))).toBe(true);
  });

  it('should have an output to match the snapshot', () => {
    plugin(mockAST, file);
    expect(file.data).toMatchSnapshot();
  });
});
