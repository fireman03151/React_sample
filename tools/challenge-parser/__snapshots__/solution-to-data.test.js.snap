// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`challengeSeed-to-data plugin should have an output to match the snapshot 1`] = `
Object {
  "solutions": Array [
    Object {
      "indexjs": Object {
        "contents": "function testFunction(arg) {
  return arg;
}

testFunction('hello');
",
        "ext": "js",
        "head": "",
        "key": "indexjs",
        "name": "index",
        "tail": "",
      },
    },
  ],
}
`;
